### PR TITLE
ST_Explode select query

### PR DESCRIPTION
**Pull request by ebocher**
The ST_Explode function supports now a SELECT query.

```mysql
SELECT * FROM ST_EXPLODE('SELECT * FROM BRETAGNE WHERE DEPARTEMENT=56');
```

Related to #418

### CODE REVIEW

**Comment by nicolas-f** [comment]
Please Fix Unit test conflict

File: h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/properties/ST_Explode.java [hunk 1]
```diff
@@ -75,6 +75,10 @@ public String getJavaStaticMethod() {
      * @throws java.sql.SQLException
      */
     public static ResultSet explode(Connection connection, String tableName) throws SQLException {
+        if(tableName.toLowerCase().startsWith("select")){
```
Comment (nicolas-f) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/properties/ST_Explode.java]: Add a space at the end. If my table name is selection for example.
```java
if(tableName.toLowerCase().startsWith("select ")){
```

File: h2spatial-ext/src/test/java/org/h2gis/h2spatialext/SpatialFunctionTest.java [hunk 2]
```diff
@@ -129,6 +129,47 @@ public void test_ST_ExplodeEmptyGeometryCollection() throws Exception {
...
+                + " boundary MULTIPOLYGON);"
+                + "INSERT INTO forests VALUES(109, 'Green Forest', ST_MPolyFromText( 'MULTIPOLYGON(((28 26,28 0,84 0,"
+                + "84 42,28 26), (52 18,66 23,73 9,48 6,52 18)),((59 18,67 18,67 13,59 13,59 18)))', 101));");
+        ResultSet rs = st.executeQuery("SELECT ST_AsText(boundary) FROM ST_Explode('select * from forests') WHERE name = 'Green Forest' and explod_id=2");
+        assertTrue(rs.next());
+        assertEquals("POLYGON ((59 18, 67 18, 67 13, 59 13, 59 18))", rs.getString(1));
+        st.execute("drop table forests");
+    }
+    
+    @Test(expected = NullPointerException.class)
```
Comment (nicolas-f) [inline h2spatial-ext/src/test/java/org/h2gis/h2spatialext/SpatialFunctionTest.java]: Your function should throw the original SQL exception in order to understand why the query fail.

File: h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/properties/ST_Explode.java [hunk 3]
```diff
@@ -222,4 +226,149 @@ public ResultSet getResultSet() throws SQLException {
             return rs;
         }
     }
```
Comment (nicolas-f) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/properties/ST_Explode.java]: :( source code duplication..

Comment (ebocher) [inline h2spatial-ext/src/main/java/org/h2gis/h2spatialext/function/spatial/properties/ST_Explode.java]: Yes. But I prefer  to have two different codes due to the aspect of the signature.

File: spatial-utilities/src/main/java/org/h2gis/utilities/SFSUtilities.java [hunk 4]
```diff
@@ -297,6 +297,44 @@ private static ResultSet getGeometryColumnsView(Connection connection,String cat
...
+    
+    /**
+     * Find the first geometry field name of a resultSet. Return null if there
+     * is no geometry column
+     *
+     * @param resultSet
+     * @return The name of the first Geometry field
+     * @throws SQLException
+     */
+    public static String getFirstGeometryFieldName(ResultSet resultSet) throws SQLException {
```
Comment (nicolas-f) [inline spatial-utilities/src/main/java/org/h2gis/utilities/SFSUtilities.java]: you should use the function getFirstGeometryFieldIndex

Comment (ebocher) [inline spatial-utilities/src/main/java/org/h2gis/utilities/SFSUtilities.java]: This function is not used anymore. I will delete it.

**Pull request merged**